package/MDAnalysis/core/universe.py

```diff
@@ -1426,82 +1445,146 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
+        **kwargs: extra arguments to be passed to the guesser class
+        
+        Examples
+        --------
+        guess masses and elements attribute::
+
+        >>>u.guess_TopologyAttributes(context='default', to_guess=['masses', elements])
+    
+
+        .. versionadded:: 2.4.0
+
+        """
+        if not context:
+            context = self._context
+
+        guesser = get_guesser(context, self.universe, **kwargs)
+        self._context = guesser
+
+        to_guess = list(to_guess)
+        force_guess = list(force_guess)
+        guess = force_guess + to_guess
+        
+        # removing duplicate from the guess list with keeping attributes order 
+        # as it is more convientent to guess attributes
+        # in the same order that the user provided
+        guess = list(dict.fromkeys(guess))
+
+        objects = ['bonds', 'angles', 'dihedrals', 'impropers']
+    
+        if(guess):
+            if (self._topology.n_atoms > 0):
+
+                if guesser.are_guessable(guess):
+
+                    for attr in to_guess:
+                        if any(attr == a for a in self._topology_atrrs):
+                            logger.info('The attribute {} have already been read '
+                                        'from the topology file. The '
+                                        'guesser will only guess empty values for this attribute, if any exists.'
+                                        'To overwrite it by completely guessed values, you can pass the attribute to the force_guess'
+                                        'parameter instead of the to_guess one'
+                                        .format(attr))
+                            
+                    for attr in guess:
+                        fg = False
+                        if attr in force_guess:
+                            fg = True
+                        values = guesser.guess_attr(attr, fg)
+                        
+                        if values is not None:
+                            if attr in objects:
+                                self._add_topology_objects(attr, values, guessed=True)
+                            else:
+                                tcls = _TOPOLOGY_ATTRS[attr](values, True)
+                                self.add_TopologyAttr(tcls)
+
+
+                else:
+                    raise ValueError('{0} guesser can not guess one or more '
+                                     'of the provided attributes'
+                                      .format(context))      
+
+            else:
+                warnings.warn('Can not guess attributes '
+                                 'for universe with 0 atoms')
+
 
 def Merge(*args):
     """Create a new new :class:`Universe` from one or more
     :class:`~MDAnalysis.core.groups.AtomGroup` instances.
-
```

Comment:
You cannot remove these line skips in the docstring: it will break the formatting of the documentation.